test/integration/backward_compatible/serial/map/MapProxyTest.js

```diff
@@ -153,6 +153,42 @@ describe('MapProxyTest', function () {
                 expect(val).to.be.true;
             });
 
+            it('basic_remove_all', async function () {
+                const predicate = await Predicates.between('this', 'val2', 'val5');
+                await map.removeAll(predicate);
+                const size = await map.size();
+                expect(size).to.equal(6);
+            });
+
+            // it('remove_all_with_null_predicate', async function () {
+            //     const predicate = await Predicates.equal('this', null);
+            //     const respond = await map.removeAll(predicate);
+            //     expect(respond).throws(AssertionError);
+            // });
```

Comment:
```suggestion
it('removeAll_throws_exception_whenPredicateNull', async function () {
    const rejectionReason = await TestUtil.getRejectionReasonOrThrow(async () => await map.removeAll(null));
    expect(rejectionReason).to.be.instanceOf(AssertionError);
});
```

please note that AssertionError needs to be imported from 'assert' package:

`const { AssertionError } = require('assert');`